### PR TITLE
Replace YAML::Tiny with YAML::PP 

### PR DESCRIPTION
**Pull request by rwx788**
#10100 was reverted due to some schedule having curly braces not wrapped in the quotes.
So I've decided to use this opportunity not only to bring the changes back, but also to use suggest way for the parser usage and not legacy interface as we did initially.

We want to replace usage of YAML::Tiny with YAML::PP, which has multiple
additional features we could benefit from multiple features instead of
implementing it ourselves.

After `os-autoinst-distri-opensuse-deps` is deployed to the workers
we can merge this PR, to be sure that dependencies are available.

[Verification run](https://openqa.suse.de/t4182047).

### CODE REVIEW

**Comment by perlpunk** [comment]
FYI, I hacked together a quick JSON schema, and all files were loaded as valid. For some reason I missed the two files that still had `{{ }}`.
Anyway, should we finish the JSON schema testing first before merging this?

**Comment by rwx788** [comment]
> FYI, I hacked together a quick JSON schema yesterday, and all files were loaded as valid. For some reason I missed the two files that still had `{{ }}`.
> Anyway, should we finish the JSON schema testing first before merging this?

I see that we don't sync our work well =(. This PR is fine to be merged anytime and we are not in the rush. So I will keep it open, and we can get JSON schema working.

**Comment by perlpunk** [comment]
I created a draft: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/10160

**Comment by rwx788** [comment]
#10160 is merged, so we can proceed with this one.

File: lib/scheduler.pm [hunk 1]
```diff
@@ -24,14 +24,17 @@ use warnings;
...
-use YAML::Tiny;
+use YAML::PP;
 use Data::Dumper;
 
 our @EXPORT = qw(load_yaml_schedule get_test_suite_data);
 
 my $test_suite_data;
 my $include_tag = '$include';
 
+my $ypp = YAML::PP->new();
```
Comment (perlpunk) [inline lib/scheduler.pm:35]: Just a note:
If we planned to enable merge keys already in this PR, the object would have to be initialized like that:
```perl
my $ypp = YAML::PP->new( schema => [qw(+ Merge )] );
```

Comment (rwx788) [reply lib/scheduler.pm:35]: Thanks for the hint, I will add this note to https://progress.opensuse.org/issues/66206 where we plan to address it and use this functionality to get rid of our self-written solution.

**Pull request merged**